src/envoy/http/authn/peer_authenticator.cc

```diff
@@ -47,7 +47,11 @@ void PeerAuthenticator::runMethod(const iaapi::PeerAuthenticationMethod& method,
                                   const MethodDoneCallback& done_callback) {
   switch (method.params_case()) {
     case iaapi::PeerAuthenticationMethod::ParamsCase::kMtls:
-      validateX509(method.mtls(), done_callback);
+      if (!method.mtls().allow_tls()) {
```

Comment:
It shouldn't fork here, better inside the validateX509 (rename it if you want) and relax the condition for not having certificate.

If fork here, we need to keep both function in sync in the future, which I think is harder to maintain.

Plus, we want to keep this validateXXX is 1-1 mapping with each method supported by policy.

Reply:
Done.